### PR TITLE
docs(ia) Improve information architecture of the Capabilities section and metadata content

### PR DESCRIPTION
**Pull request by pwoznic**
Made several improvements to the Capabilities section:

- Divided and chunked the content into new sections for Data Management, Search and Retrieval, and Generation and Observability
- Refined the previous Metadata Filters section by chunking into separate child topics, improving navigation and usability
- Miscellaneous improvements throughout these new sections

### CODE REVIEW

**Comment by cjcenizal** [comment]
@pwoznic I love this new organization! Thanks for doing this. One suggestion: can we split "Generation and Observability" into "Generation" and "Observability and Evaluation"? I think the latter stands alone, as it's a major area in which we're investing.

The hierarchy would then be:
* Generation
  * Mockingbird LLM
  * Summarizers
  * Generative prompts
* Observability and Evaluation
  * Query Observability
  * Hallucination Evaluation

Also, on a separate note, "summarizers" no longer exist in API v2 as a concept. They're now replaced by "generation presets". Per the Query API docs:
image

File: www/docs/generation.mdx [hunk 1]
```diff
@@ -1,17 +1,18 @@
...
 import { Spacer } from "@site/ui/Spacer";
 import { TopicButton } from "@site/ui/TopicButton";
 
-Leverage Vectara’s advanced capabilities for content generation and quality 
-assurance. This section ensures that your AI-driven outputs are accurate, 
-relevant, and aligned with your goals.
+Generative AI empowers organizations to create precise, context-aware outputs 
+from their data. Vectara offers advanced tools to help you craft tailored 
+responses, summaries, and insights. These capabilities improve accuracy, and 
+enhance the value of your AI-driven applications.
```
Comment (omnomagonz) [inline www/docs/generation.mdx:15]:
```suggestion
Generative AI, within the RAG context, empowers organizations to create precise,
context-aware outputs from their data. Vectara offers advanced tools to help you
craft tailored responses, summaries, and insights. These capabilities improve
accuracy, and enhance the value of your AI-driven applications.
```

Non-blocking: could it be more apt to frame it this way? The link to "... from their data" to me seems more like a RAG-specific thing vs GenAI in general.

File: www/docs/learn/grounded-generation/summarizers.md [hunk 2]
```diff
@@ -37,28 +42,28 @@ summaries and structured outputs:
...
+gpt-4.0-turbo.
 
 :::caution
-The Vectara trial includes access to all GPT4-based summarizers. After the 
-trial ends and you upgrade your plan, you can purchase separate GPT4 bundles 
-or bring your own API key. Vectara bundles offer the advantage of HIPAA 
-compliance.
+The Vectara trial includes access to all GPT4-based generation presets. After 
+the trial ends and you upgrade your plan, you can purchase separate GPT4 
+bundles or bring your own API key. Vectara bundles offer the advantage of
```
Comment (omnomagonz) [inline www/docs/learn/grounded-generation/summarizers.md:61]:
```suggestion
bundles or bring your own API key. Vectara bundles offer the advantage of 
```

Do we support BYOKey? For some reason I thought we didn't formally support this yet but I may be mistaken

**Review by omnomagonz — APPROVED** [review]

File: www/docs/learn/grounded-generation/summarizers.md (hunk 2 above)
Comment (pwoznic) [reply www/docs/learn/grounded-generation/summarizers.md:61]: Yes, I double-checked our pricing page and it says there are bundles or BYOK.

**Pull request merged**